daliuge-engine/dlg/manager/proc_daemon.py

```diff
@@ -104,22 +107,44 @@ def __init__(self, master=False, noNM=False, disable_zeroconf=False, verbosity=0
         app.get("/managers/node", callback=self.rest_getNMInfo)
 
         # Automatically start those that we need
+        if master and dim:
+            self._zeroconf = False
         if master:
             self.startMM()
-        if not noNM:
+        if nm:
             self.startNM()
+        if dim:
+            nodes = []
+            if nm:
+                # addrs = utils.get_local_ip_addr()
+                addrs = "127.0.0.1"
+                nm_port = constants.NODE_DEFAULT_REST_PORT
+                nodes = [f"{addrs}:{nm_port}"]
```

Comment:
I don't think we need 4lines for this, as it makes the code look like its doing more than it is:

```
if nm:
    nodes.append([f"127.0.0.1:{constants.NODE_DEFAULT_REST_PORT}}])
```